week3/runoff.c

```diff
@@ -0,0 +1,210 @@
+#include "cs50.h"
+#include <stdio.h>
+#include <string.h>
+
+#define MAX_VOTERS 100
+#define MAX_CANDIDATES 9
+
+int preferences[MAX_VOTERS][MAX_CANDIDATES];
+
+typedef struct
+{
+    string name;
+    int votes;
+    bool eliminated;
+}
+candidate;
+
+candidate candidates[MAX_CANDIDATES];
+int voter_count;
+int candidate_count;
+
+bool vote(int voter, int rank, string name);
+void tabulate(void);
+bool print_winner(void);
+int find_min(void);
+bool is_tie(int min);
+void eliminate(int min);
+
+int main(int argc, string argv[])
+{
+    if (argc < 2)
+    {
+        printf("Usage: runoff [candidate ...]\n");
+
+        return 1;
+    }
+
+    candidate_count = argc - 1;
+
+    if (candidate_count > MAX_CANDIDATES)
+    {
+        printf("Maximum number of candidates is %i\n", MAX_CANDIDATES);
+
+        return 2;
+    }
+
+    for (int candidate_number = 0; candidate_number < candidate_count; candidate_number++)
+    {
+        candidates[candidate_number].name = argv[candidate_number + 1];
+        candidates[candidate_number].votes = 0;
+        candidates[candidate_number].eliminated = false;
+    }
+
+    voter_count = get_int("Number of voters: ");
+
+    if (voter_count > MAX_VOTERS)
+    {
+        printf("Maximum number of voters is %i\n", MAX_VOTERS);
+
+        return 3;
+    }
+
+    for (int voter_number = 0; voter_number < voter_count; voter_number++)
+    {
+        for (int candidate_number = 0; candidate_number < candidate_count; candidate_number++)
+        {
+            string name = get_string("Rank %i: ", candidate_number + 1);
+
+            if (!vote(voter_number, candidate_number, name))
+            {
+                printf("Invalid vote.\n");
+
+                return 4;
+            }
+        }
+
+        printf("\n");
+    }
+
+    while (true)
+    {
+        tabulate();
+
+        bool won = print_winner();
+        if (won)
+        {
+            break;
+        }
```

Comment:
```suggestion
        if (won) return 0;
```
Wir möchten ja, wenn jemand gewonnen hat, dass das Programm den Gewinner ausgibt und sich mit einem success value beendet, was meine ich 0 war.